app-android-xamarin/Moodis/Moodis/Feature/Group/GroupListActivity.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Support.V7.App;
+using Android.Support.V7.Widget;
+using Android.Views;
+using Android.Widget;
+using Moodis.Extensions;
+
+namespace Moodis.Feature.Group
+{
+    [Activity(Label = "Group List")]
+    public class GroupListActivity : AppCompatActivity
+    {
+        private RecyclerView recyclerView;
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+            this.SetSupportActionBar();
+            SetContentView(Resource.Layout.activity_group_list);
+            InitAdapter();
+        }
+
+        public override bool OnSupportNavigateUp()
+        {
+            OnBackPressed();
+            return true;
+        }
```

Comment:
Insert Finish() to close the activity. Or just remove the override completely. The default OnSupportNavigateUp() tries to finish the activity and if it succeeds it returns true, now you are removing it's functionality and asking it to return you to a previous activity.